### PR TITLE
Update docs, add FAQ 

### PR DESCRIPTION
**Pull request by trevorsibanda**
I think this would help alot of people new to Slick.

### CODE REVIEW

**Comment by cvogt** [comment]
LGTM. @d6y ?

File: slick/src/sphinx/faq.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+This page contains a list of frequently asked questions and other gotchas to help you get started with Slick. 
+Be sure to read the :doc:`Getting Started <gettingstarted>` page for more info. 
+
+
+General
+-------
+
+* What is Slick ?
+
+Slick ("Scala Language-Integrated Connection Kit") is `Typesafe <http://www.typesafe.com>`_'s
```
Comment (lastland) [inline slick/src/sphinx/faq.rst]: Typesafe has been renamed as Lightbend. Probably should use the new name here?

File: slick/src/sphinx/faq.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Functional Relational Mapping (FRM) library for Scala that makes it easy to work with relational
+databases. It allows you to work with stored data almost as if you were using Scala collections
+while at the same time giving you full control over when database access happens and which data
+is transferred. You can also use SQL directly. Execution of database actions is done
+asynchronously, making Slick a perfect fit for your reactive applications based on Play_ and Akka_.
+
+* How do I get Slick ?
```
Comment (d6y) [inline slick/src/sphinx/faq.rst]: Unless we answer the question, I think it'd be better to remove this heading (e.g., for now).

**Comment by d6y** [comment]
@trevorsibanda thank you - having a place to put FAQs will be great.  Take a look at the above two commands, and see what you think.

**Comment by cvogt** [comment]
@trevorsibanda no need to close a PR if you are still working on it :)

**Comment by trevorsibanda** [comment]
@cvogt I deleted my fork, the one that had the PR.

I made some mistakes with the branches, tried rebasing. I have all the changes and more on my local repo.

**Pull request closed without merging**